app/core/models.py

```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class Document:
+    id: int | None
+    project_id: int
+    upload_filename: str
+
+
+@dataclass
+class User:
+    id: int | None
+    username: str
+    password: str
+
+
+@dataclass
+class Project:
+    id: int | None
+    name: str
+    description: str
+    owner: str
+    collaborators: set[str]
```

Comment:
it could be collaborators: set[User] or collaborators: List[User], instead of collaborators: set[str]. User would be more descriptive and it is allowed in domain. str is ambigous because it could be caseted id, name ect...

Reply:
```suggestion
    owner_usernames: str
    collaborator_usernames: set[str]
```
I think that OO approach is not needed here, as in project context we're only interested with usernames (for permission), not the whole User model. These field names should be more precise.

This solution avoids creating additional objects(Users) and I think is sufficient for our simple domain.